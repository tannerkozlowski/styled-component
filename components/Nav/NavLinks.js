import React from 'react'
import styled from 'styled-components'

import rem from '../../utils/rem'
import { navbarHeight } from '../../utils/sizes'
import NavSeparator from './NavSeparator'
import Link from '../Link'

const Wrapper = styled.nav`
  display: flex;
  align-items: center;
`

const NavLink = styled(Link).attrs({
  unstyled: true,
})`
  display: inline-block;
  line-height: ${rem(navbarHeight)};
  transition: opacity 200ms, transfrom 200ms;
  cursor: pointer;

  letter-spacing: ${rem(.4)}
  color: currentColor;

  &:hover,
  &:focus {
    opacity: 0.8;
  }

  &:active {
    transform: scale(0.95);
    opacity: 0.6;
  }
`

const NavLinks = () => (
  <Wrapper>
    <NavLink>Concepts</NavLink>
    <NavSeparator />
    <NavLink>Try it out</NavLink>
    <NavSeparator />
    <NavLink>Documentation</NavLink>
  </Wrapper>
)

export default NavLinks
