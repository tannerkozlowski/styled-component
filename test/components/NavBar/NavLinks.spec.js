import React from 'react'
import renderer from 'react-test-renderer'
import 'jest-styled-components'
import NavLinks from '../../../components/Nav/NavLinks'

test('NavLinks renders correctly', () => {
  const tree = renderer.create(<NavLinks />).toJSON()

  expect(tree).toMatchSnapshot()
})
