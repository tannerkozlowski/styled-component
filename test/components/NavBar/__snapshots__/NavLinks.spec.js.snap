// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavLinks renders correctly 1`] = `
<nav
  className="eWokkx"
>
  <a
    className="fAKxmD"
    href="/docs"
    onClick={[Function]}
  >
    Documentation
  </a>
  <span
    className="lnegGh"
  />
  <a
    className="fAKxmD"
    href="/ecosystem"
    onClick={[Function]}
  >
    Ecosystem
  </a>
  <span
    className="lnegGh"
  />
  <a
    className="fAKxmD"
    href="/releases"
    onClick={[Function]}
  >
    Releases
  </a>
</nav>
`;
